storage/boltz/store_crud.go

```diff
@@ -198,7 +206,16 @@ func (store *BaseStore) Update(ctx MutateContext, entity Entity, checker FieldCh
 	entity.SetValues(persistCtx)
 	indexingContext.ProcessAfterUpdate() // add new values, using updated values in store
 
-	ctx.AddEvent(store, EventUpdate, entity)
+	// for consistent ordering we do this before update handlers,
+	// since child events will be generated there. It's ok to do
+	// this early as if there is an error, the events won't be
+	// propagated
+	store.AddEvent(ctx, entity, EventUpdate)
+
+	if err := store.updateHandlers.Handle(ctx, entity.GetId()); err != nil {
+		return err
+	}
+
```

Comment:
So the order of events is:

1) Top-level entity is updated/created/deleted
2) Child entity is update/created/deleted

If so then for update, to get the final "updated" entity you have to listen for the child updates?

Reply:
You can listen for either. The parent store will get parent entities and the child store will get child entities. Either way they should be up to date.

Reply:
Oh I see, gotcha.